### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by lightz96**
1. Insert sections from external files to DG (SettingUp, Documentation, Testing, Dev Ops) 
2. Update Undo and redo implementation images.
3. Update the UI class diagram and Logic class diagram.

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -36,19 +36,95 @@ This developer guide specifies the technical and non-technical details of THRIFT
...
+* For `Class count to use import with '\*'` and `Names count to use static import with '*'`: Set to *999* to prevent IntelliJ from contracting the import statements
+* For `Import Layout`: The order is *import static all other imports*, *import java.\**, *import javax.**, *import org.\**, *import com.**, *import all other imports*. Add a *<blank line>* between each *import*
+
+Optionally, you can follow the <<UsingCheckstyle#, UsingCheckstyle.adoc>> document to configure Intellij to check style-compliance as you write code.
+
+==== Updating documentation to match your fork
+
+After forking the repo, the documentation will still have the SE-EDU branding and refer to the _se-edu/addressbook-level3_ repo.
+
+If you plan to develop this fork as a separate product (i.e. instead of contributing to _se-edu/addressbook-level3_), you should do the following:
```
Comment (limerencee) [inline docs/DeveloperGuide.adoc]: I think we might need to change this to THRIFT context. If the developer is to continue building onto our application.

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -36,19 +36,95 @@ This developer guide specifies the technical and non-technical details of THRIFT
...
+WARNING: This symbol indicates information that you need to be cautious of when enhancing THRIFT.
 
 [horizontal]
 *add_expense n/Laksa v/4*:: Bold text indicates user input.
 
-_/src/java/thrift/logic/Logic.java_:: Italic text indicates that it is a file path.
+_/src/java/thrift/logic/Logic.java_:: Italic text indicates that it is a file/github path.
 
-`currentMonth`:: Text with grey highlight (called a mark-up) indicates that it is a method/variable name.
+`currentMonth`:: Text with grey highlight (called a mark-up) indicates that it is a method/variable/setting parameter name.
```
Comment (limerencee) [inline docs/DeveloperGuide.adoc]: Maybe just "parameter" and not "setting parameter"

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -803,15 +879,188 @@ instead of just its size
...
+|_not set_
+
+|`site-githuburl`
+|URL to the site's repository on https://github.com[GitHub].
+Setting this will add a "View on GitHub" link in the navigation bar.
+|_not set_
+
+|`site-seedu`
+|Define this attribute if the project is an official SE-EDU project.
+This will render the SE-EDU navigation bar at the top of the page, and add some SE-EDU-specific navigation items.
```
Comment (limerencee) [inline docs/DeveloperGuide.adoc]: Might need to change to THRIFT context as well

**Review by limerencee — CHANGES_REQUESTED** [review]
LGTM after trivial changes

File: docs/DeveloperGuide.adoc (hunk 3 above)
Comment (lightz96) [reply docs/DeveloperGuide.adoc]: This attribute `site-seedu` is meant for SE_EDU. Maybe I should remove this attribute from DG.

**Review by limerencee — APPROVED** [review]
LGTM

**Pull request merged**